third_party/ienn/src/ie_execution.cpp

```diff
@@ -160,37 +160,42 @@ int32_t Execution::StartCompute() {
     }
 
     infer_request->Infer();
-
+    OutputsDataMap outputInfo = compilation_->network_->getOutputsInfo();
+    auto outputInfoItem = outputInfo.begin();
     for (size_t i = 0; i < model->outputs.size(); ++i) {
       uint32_t index = model->outputs[i];
       const Operand& operand = model->operands[index];
       const uint32_t offset = total_length;
       const uint32_t length = output_data_[i].length;
-      ;
       total_length += length;
       void* mapping = output_data_[i].buffer;
 
-      OutputsDataMap outputInfo = compilation_->network_->getOutputsInfo();
-      auto outputInfoItem = outputInfo.begin();
       Blob::Ptr output_blob = infer_request->GetBlob(outputInfoItem->first);
-      outputInfoItem ++;
       if (operand.dimensions.size() == 3) {
         const float* src =
             output_blob->buffer()
                 .as<PrecisionTrait<Precision::FP32>::value_type*>();
+        std::vector<unsigned int> dimensions(operand.dimensions);
+        if (outputInfoItem->first.find("TopK") != std::string::npos) {
```

Comment:
Can we get the shape with `const Shape& get_output_shape(size_t i) const`; from Node? it's calcuated, not set by JavaScript.

Reply:
Yes, but the layout of the node is NCHW, and the second parameter `dimension` of `Reorder` function show define the operand dimension of the model set by javascript.

Reply:
We can convert it to NHWC if it's our need, and the output dimensions will be remove in new spec.

Reply:
We suppose we don't need it in this situation. We define the `dimensions` to  pass prammeter for `Reorder` function.  The second parameter only describes the dimensions of the operand of model set from js.

Reply:
Do you mean the specical case can be handled in polyfill?

Reply:
No, it can't be handle in polyfill since other backends all support 3D.  We need to handle this special case from IE.

Reply:
We should investigate the reason of reordering output when the layout is set to [NHWC](https://github.com/otcshare/chromium-src/blob/webml/third_party/ienn/src/ie_compilation.cpp#L136), and can we use `const Shape& get_output_shape(size_t i) const` instead of ` operand.dimensions` if it's need to reorder?

Reply:
The output layout is still **NCHW** although we have set to **NHWC** , so reordering the output can get correct result. Should we delete the `set to **NHWC**` and reorder the output when dimension.size() equals to 3 or 4?

Reply:
So we should get the layout of output node like `if (node->getLayout() == "NCHW")` instead of `if (operand.dimensions.size() == 3)`, and convert `{1,1,224,224}` NCHW to NHWC instead of `operand.dimensions`.